### PR TITLE
Reload systemctl service on updates

### PR DESCRIPTION
**Pull request by atoomic**
Fix GH #8922

### CODE REVIEW

File: builder-support/specs/dnsdist.spec [hunk 1]
```diff
@@ -153,6 +153,10 @@ fi
 %endif
 %if 0%{?rhel} >= 7
 %systemd_post %{name}.service
+if [ $1 -eq 2 ] ; then
```
Comment (pieterlexis) [inline builder-support/specs/dnsdist.spec]: I think this should come before `%systemd_post %{name}.service`

Comment (atoomic) [reply builder-support/specs/dnsdist.spec]: note that order does not really matters as long as we are using a check for update as `%systemd_post` translate to `if [ $1 -eq 1 ] ; then ... fi`
I was assuming `preset` was implying a daemon-reload but after reading some docs https://freedesktop.org/wiki/Software/systemd/Preset/ it s mainly going to trigger enable 
so we would have to move it above and remove the condition

**Comment by atoomic** [comment]
probably better to just do `systemctl daemon-reload ||:` otherwise it will be impossible to know if it fails and adjust it if needed...

**Comment by phonedph1** [comment]
So in EL7 this happens in the postun triggers, and now in EL8 they are removed. Does it make sense to duplicate the same effort in the postun sections instead, to match what EL7 does?

**Comment by atoomic** [comment]
@phonedph1 I think it s a pretty bad idea to use the `postun` macro during post, as you have no guarantee that this will not change overtime/distro... and trigger some unexpected behavior

**Pull request merged**